android/src/main/java/com/clevertap/react/CleverTapListenerProxy.kt

```diff
@@ -0,0 +1,224 @@
+package com.clevertap.react
+
+import android.util.Log
+import com.clevertap.android.sdk.CTFeatureFlagsListener
+import com.clevertap.android.sdk.CTInboxListener
+import com.clevertap.android.sdk.CleverTapAPI
+import com.clevertap.android.sdk.InAppNotificationButtonListener
+import com.clevertap.android.sdk.InAppNotificationListener
+import com.clevertap.android.sdk.InboxMessageButtonListener
+import com.clevertap.android.sdk.InboxMessageListener
+import com.clevertap.android.sdk.PushPermissionResponseListener
+import com.clevertap.android.sdk.SyncListener
+import com.clevertap.android.sdk.displayunits.DisplayUnitListener
+import com.clevertap.android.sdk.displayunits.model.CleverTapDisplayUnit
+import com.clevertap.android.sdk.inapp.CTInAppNotification
+import com.clevertap.android.sdk.inbox.CTInboxMessage
+import com.clevertap.android.sdk.product_config.CTProductConfigListener
+import com.clevertap.android.sdk.pushnotification.CTPushNotificationListener
+import com.facebook.react.bridge.Arguments
+import org.json.JSONException
+import org.json.JSONObject
+
+object CleverTapListenerProxy : SyncListener, InAppNotificationListener, CTInboxListener,
+    InboxMessageButtonListener, InboxMessageListener, InAppNotificationButtonListener,
+    DisplayUnitListener, CTProductConfigListener, CTFeatureFlagsListener,
+    CTPushNotificationListener, PushPermissionResponseListener {
+
+    private const val LOG_TAG = Constants.REACT_MODULE_NAME
+
+    fun attachToInstance(instance: CleverTapAPI) {
+        instance.registerPushPermissionNotificationResponseListener(this)
+        instance.ctPushNotificationListener = this
+        instance.inAppNotificationListener = this
+        instance.syncListener = this
+        instance.ctNotificationInboxListener = this
+        instance.setInboxMessageButtonListener(this)
+        instance.setCTInboxMessageListener(this)
+        instance.setInAppNotificationButtonListener(this)
+        instance.setDisplayUnitListener(this)
+        instance.setCTProductConfigListener(this)
+        instance.setCTFeatureFlagsListener(this)
```

Comment:
with (instance) { }

Reply:
In this case, I think, ```with(instance)``` looks clunkier, since ```this``` is captured by the receiver and the listener would have to be labeled with ```this@CleverTapListenerProxy```. It would look like this:
```
with(instance) {
    registerPushPermissionNotificationResponseListener(this@CleverTapListenerProxy)
}
```

Overall not a big difference, we could do it either way.